### ISSUE TITLE
Support for adding ion binding-energy equation terms 

### ISSUE THREAD

**Issue by Embreus** [chalmersplasmatheory/DREAM#55]
We should implement support to 'automatically' add the contribution due to binding energy in the energy equation. Background:

The ion equation 
dn_i/dt = IonRateEquation + sum_k (dn_i/dt)_k
where (dn_i/dt)_k are a set of equation terms that affect the evolution of ions, such as transport, ionization by fast electrons, ablation of pellets and possibly others in the future.
 
The energy equation is
dW/dt = ...
where 
W = W_binding + 1.5nT,
where 
W_binding = sum_i E_i*n_i,
with E_i the binding potentials of each charge state.

The reason that W is split up like this is because the binding-energy contribution to the energy equation, dW_binding/dt, due to the IonRateEquation is quite complex but is provided directly by the ADAS energy rate coefficients (essentially the recombination radiation), so that it is convenient to lump W_binding and nT together in this manner.

The issue now is that we must add the contribution dW_binding/dt from all the other ion equation terms (dn_i/dt)_k, which is not done properly yet. We should implement some method which sums over all `EquationTerm`s except for the `IonRateEquation` and `TransientTerm`, and adds a combined term
S = sum_k sum_i E_i (dn_i/dt)_k
to the energy equation for dW/dt. 

Open to ideas of how this could be done in an elegant way.

**Comment by Embreus** [comment]
A maybe-not-so-wild idea. We could change W_cold to be simply 1.5nT, and then only add the ion binding rate due to IonRateEquation. This would require much less (no) maintenance in the future.

**Comment by Embreus** [comment]
I have solved this issue on 83304a6, by following the suggestion in the previous comment. The binding-energy change due to IonRateEquation has been added into the RadiatedPowerTerm, which now represents the net energy loss due to ncold-ncold atomic physics. 

It will now be more clear how to add terms to the ion equation. If we don't want to explicitly heat or cool the cold-electron population, such as by adding new ions or by transporting ions, we do not need to add counter-terms to the heat equation like we did before.
